### PR TITLE
remove all leftover data-hooks

### LINKED ISSUE #1586: Rip out all of the data-hooks
I don't think this is an issue yet...

Lots of legacy hooks from our Ampersand days. Let's rip them out.

nb. there's one CSS reference to a data-hook, so make sure to handle that.

### PR DESCRIPTION
**Pull request by chuckharmston**
Copying from https://github.com/mozilla/testpilot/pull/1933 to diagnose missing CI.

---

I got really sick of looking at these things. Fixes #1586

NB. There are two places where these stayed in: 
- React uses a data-hook to mount the app
- in the email component, a data-hook is used in a test

### CODE REVIEW

File: frontend/src/app/components/EmailDialog.js
Comment (jaredhirsch) [inline frontend/src/app/components/EmailDialog.js]: How about an `id="second-page"` for readability? (I think the ID would help more here than in the first-page case, but for symmetry, might be better to keep or lose both.)

File: frontend/src/app/components/EmailDialog.js
Comment (jaredhirsch) [inline frontend/src/app/components/EmailDialog.js]: It looks like the data-hook slipped by here. Please remove it.

File: frontend/src/app/components/EmailDialog.js
Comment (jaredhirsch) [inline frontend/src/app/components/EmailDialog.js]: How about an `id="first-page"` for readability?

File: frontend/src/app/components/ExperimentRowCard.js
Comment (jaredhirsch) [inline frontend/src/app/components/ExperimentRowCard.js]: `id="show-detail"`?

File: frontend/src/app/components/Header.js
Comment (jaredhirsch) [inline frontend/src/app/components/Header.js]: `id="settings"`?

File: frontend/src/app/containers/ExperimentPage.js
Comment (jaredhirsch) [inline frontend/src/app/containers/ExperimentPage.js]: `id="testpilot-promo"`?

File: frontend/src/app/containers/ExperimentPage.js
Comment (jaredhirsch) [inline frontend/src/app/containers/ExperimentPage.js]: `id="details"`?

File: frontend/src/app/containers/HomePageNoAddon.js
Comment (jaredhirsch) [inline frontend/src/app/containers/HomePageNoAddon.js]: `id="landing-page"`?

File: frontend/src/app/containers/NotFoundPage.js
Comment (jaredhirsch) [inline frontend/src/app/containers/NotFoundPage.js]: `id="not-found-page"`?

**Review by jaredhirsch — CHANGES_REQUESTED** [review]
Hey @johngruen - Looks good overall. Couple minor things:

First, if we're going to this much trouble to remove data-hooks, I think it makes sense to remove the last two occurrences as well: a selector that React can use to bootstrap the app, and a selector used by a test. I bet that replacing the data-hook with an ID would work in both cases. I do think these last two data-hooks should be removed before landing this PR.

Second, in some cases, I think it'd be good to keep the data-hook's content, just to keep things readable/scannable. Especially when the data-hook was in the topmost node returned by a component, it's just nice to have when scanning the DOM later. I've called out spots where I think this change would help, but because readability is more of a judgment call, consider those comments to be suggestions--feel free to disregard if you don't agree.

Feel free to merge this yourself after making changes, or ping me if you want a followup review. Thanks!

**Comment by johngruen** [comment]
@6a68 yeah good points above. I'll work back through this after we land the various last minute l10n issues today

**Comment by johngruen** [comment]
@6a68 changes made. Could you give it one more look and merge since i touched a lot of files.

Cheers!

**Comment by johngruen** [comment]
Rebased!

**Comment by jaredhirsch** [comment]
lgtm, merging :+1:

**Pull request merged**
